### PR TITLE
Server-side Scribe rendering support

### PR DESCRIPTION
**Pull request by rrees**
Combines #438 and #436 with a unit-testing strategy that might help us get around the flakiness in the browser driven tests.

@szyablitsky  @SimonDegraeve can you check that this code matches your intent?

### CODE REVIEW

**Comment by SimonDegraeve** [comment]
It sounds good to me.

**Comment by szyablitsky** [comment]
LGTM :+1:

File: src/mutations.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+define([], function() {
+
+  function determineMutationObserver(window) {
```
Comment (szyablitsky) [inline src/mutations.js]: @rrees Server side rendering still does not work.
You pass the `window` to the function. And `window` is not defined on server.
I guess you added this parameter only for testing purposes. But it nullifies the intent of this patch.
May be you can mock window object inside the test? So we do not need to pass it as a parameter.

Comment (rrees) [inline src/mutations.js]: No this one is meant to be a parameter, I'll look at what we can do in the caller

File: src/dom-observer.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-  './node'
-], function (nodeHelpers) {
+  './node',
+  './mutations'
+], function (nodeHelpers, mutations) {
 
-  var MutationObserver = window.MutationObserver ||
-    window.WebKitMutationObserver ||
-    window.MozMutationObserver;
+  var MutationObserver = mutations.determineMutationObserver(window);
```
Comment (szyablitsky) [inline src/dom-observer.js]: @rrees here you go. Calling the function with `window` as a param.

**Pull request merged**